mcs/class/System/System.Net.Sockets/SafeSocketHandle.cs

```diff
@@ -37,6 +37,7 @@ protected override bool ReleaseHandle ()
 			int error = 0;
 
 			Socket.Blocking_internal (handle, false, out error);
+			Socket.Close_internal (handle, out error);
```

Comment:
This change is wrong, we must first abort the threads then close the socket, otherwise there's a window where a new fd can be opened and the target threads would wrongly use it.